### PR TITLE
Recent repo search

### LINKED ISSUE #9302: Recent repository search
## Feature description
Background: I'm working with a lot of different repositories and I need to switch between quite regulary / fast.
It would be great to see some sort of recent repository search where I can just type some part of the path and git extensions will show me a list of all matching repos. Ideally it should be possible to just start typing after opening git extensions.

Maybe just a big searchbar at the top of the start screen?
image

Let me know what you think

### PR DESCRIPTION
**Pull request by davidgiga1993**
Fixes #9302

## Proposed changes

- A new searchbar for filtering the recent / favorite repositories
   - Pressing return will open the first search result
- The default focus has been changed to focus the search textbox once the dashboard gets visible
   -  This allows immediate search when opening gitextensions or when going back (strg+w) to the dashboard
- A new interface has been added for filtering repositories

### Before
image

### After

image

## Test methodology 

- Manual tests:
  - Search for repo, press return
  - Search for random text (no search results), press return
  - Search for non existing repo, press return

## Test environment(s) 

- Windows 10
- UI Scaling 100%

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by drewnoakes** [comment]
This is a feature I've wanted for ages. Thank you for doing this ❤️ 

Two points:

1. What is the "Actions" text on the right?
2. Can we get a little more space above the text box so it doesn't look as cramped?

image

**Comment by pmiossec** [comment]
>     1. What is the "Actions" text on the right?

Not part of this PR. It's a recent PR #9371 done by @RussKie

**Comment by davidgiga1993** [comment]
> 2. Can we get a little more space above the text box so it doesn't look as cramped?

Like this?

image

**Comment by drewnoakes** [comment]
That does look better. Personally I would increase the space above and below to match the space to the left:

image

This follows more of a "grid system" (a concept in graphic design for creating order in a layout):

image

**Comment by davidgiga1993** [comment]
Thanks for the feedback! Let's try this again 😄 
image

**Comment by drewnoakes** [comment]
LGTM. To get the build working you need to updated the translations as described in https://github.com/gitextensions/gitextensions/wiki/Translations

**Comment by davidgiga1993** [comment]
@drewnoakes Doesn't seem to work? Followed the documentation:
1) Run `dotnet build /p:RunTranslationApp=true`
2) Commit the `English.xlf`

File: GitUI/Translation/English.xlf [hunk 1]
```diff
@@ -8709,6 +8709,14 @@ This will just checkout on the submodule the commit determined by the superproje
...
     </body>
   </file>
+  <file datatype="plaintext" original="Plugin" source-language="en">
+    <body>
+      <trans-unit id="Description.Text">
+        <source>Plugin Manager</source>
+        <target />
+      </trans-unit>
+    </body>
+  </file>
```
Comment (RussKie) [inline GitUI/Translation/English.xlf:8719]: This must not be committed, it is noted in https://github.com/gitextensions/gitextensions/wiki/Translations

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.Designer.cs [hunk 2]
```diff
@@ -164,74 +204,92 @@ private void InitializeComponent()
...
+            this.imageList1.Images.SetKeyName(0, "source_code.png");
+            // 
+            // textBoxSearch
+            // 
+            this.textBoxSearch.Dock = System.Windows.Forms.DockStyle.Fill;
+            this.textBoxSearch.Location = new System.Drawing.Point(3, 3);
+            this.textBoxSearch.Name = "textBoxSearch";
+            this.textBoxSearch.Size = new System.Drawing.Size(439, 20);
+            this.textBoxSearch.TabIndex = 0;
+            this.textBoxSearch.PlaceholderText = "Search";
```
Comment (pmiossec) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.Designer.cs:234]: I think this string should be set in the constructor and using the `TranslationString` so that the string could be translated.

Also, I don't know what other think but is "Search a repository" clearer?

**Comment by pmiossec** [comment]
@davidgiga1993 I see that you did a sync merge with `master` branch.
We try to avoid that and sync with a rebase.

Do you think you could squash and rebase on top of `master` to keep only 2 commits (the one for the contribution file and the one with your changes). If you could push the contribution file change in another PR, it's even better (that's what we advise).

I can help in doing it if you don't know how to do it...

**Comment by pmiossec** [comment]
@davidgiga1993 The easiest sometimes to solve this situation is in fact to do a 'reset soft' on the commit you synced/merged from `master` and recreate the commits like that:

GE_reset_soft_gif

After that you could compare the local branch and the remote to ensure that you don't have differences before force (with lease) pushing your branch.

**Comment by RussKie** [comment]
We have a rebase guide in the wiki.

File: GitUI/Translation/English.xlf (hunk 1 above)
Comment (davidgiga1993) [reply GitUI/Translation/English.xlf:8719]: The wiki states that this only applies to `English.Plugins.xlf` and not `English.xlf`, or am I missing something here?

**Comment by davidgiga1993** [comment]
> @davidgiga1993 I see that you did a sync merge with `master` branch.
> We try to avoid that and sync with a rebase.
> 
> Do you think you could squash and rebase on top of `master` to keep only 2 commits (the one for the contribution file and the one with your changes). If you could push the contribution file change in another PR, it's even better (that's what we advise).
> 
> I can help in doing it if you don't know how to do it...

Sure, no problem! The contrib PR is [here](https://github.com/gitextensions/gitextensions/pull/9384)

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.resx [hunk 3]
```diff
@@ -117,18 +117,18 @@
...
-    <value>166, 17</value>
+    <value>422, 17</value>
   </metadata>
   <data name="imageList1.ImageStream" mimetype="application/x-microsoft.net.object.binary.base64">
     <value>
         AAEAAAD/////AQAAAAAAAAAMAgAAAFdTeXN0ZW0uV2luZG93cy5Gb3JtcywgVmVyc2lvbj00LjAuMC4w
         LCBDdWx0dXJlPW5ldXRyYWwsIFB1YmxpY0tleVRva2VuPWI3N2E1YzU2MTkzNGUwODkFAQAAACZTeXN0
         ZW0uV2luZG93cy5Gb3Jtcy5JbWFnZUxpc3RTdHJlYW1lcgEAAAAERGF0YQcCAgAAAAkDAAAADwMAAACy
-        EAAAAk1TRnQBSQFMAwEBAAFoAQABaAEAASABAAEgAQAE/wERAQAI/wFCAU0BNgcAATYDAAEoAwABgAMA
+        EAAAAk1TRnQBSQFMAwEBAAHwAQAB8AEAASABAAEgAQAE/wERAQAI/wFCAU0BNgcAATYDAAEoAwABgAMA
```
Comment (pmiossec) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.resx:131]: That's strange. Do you know why this image has changed?

Comment (davidgiga1993) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.resx:131]: No, I thought it's just the VS forms editor doing it's thing. But it's not an intentional change.

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs [hunk 4]
```diff
@@ -8,12 +8,49 @@
...
+        {
+            if (_pattern.Trim().Length == 0)
+            {
+                return;
+            }
+
+            for (var i = repositories.Count - 1; i >= 0; i--)
+            {
+                var repo = repositories[i];
+                if (repo.Path.ToLower().Contains(_pattern))
```
Comment (pmiossec) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs:38]: For me, it's better to lower none of the strings (even the pattern above) and instead use `StringComparison.OrdinalIgnoreCase`

https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices-strings

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs [hunk 5]
```diff
@@ -8,12 +8,49 @@
...
+    /// <summary>
+    /// Case insensitive text search based on the repository path
+    /// </summary>
+    public class RepositoryTextFilter : IRepositoryFilter
+    {
+        private readonly string _pattern;
+
+        public RepositoryTextFilter(string pattern)
+        {
+            _pattern = pattern;
```
Comment (pmiossec) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs:25]: Sorry but looking at it again, it seems weird to me...
As long as you trim the string to check the length, the `.Trim()` must be here, no!?!

Either we don't trim or we use the trimmed string as the filter pattern, right?

Comment (davidgiga1993) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs:25]: Yes, you're right. It probably would make sense to trim the spaces only from the left (to ignore accidentially added spaces at the beginning of the input) but keep spaces on the right since it's legitim to have a repo path with spaces in it

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs (hunk 4 above)
Comment (drewnoakes) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs]: The comparison should probably use the machine's locale. Not everyone uses Latin characters. If suggest using StringComparison.CurrentCultureIgnoreCase.

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.Designer.cs (hunk 2 above)
Comment (drewnoakes) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.Designer.cs]: "Search repositories" seems more idiomatic to me.

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.Designer.cs [hunk 6]
```diff
@@ -136,16 +128,64 @@ private void InitializeComponent()
             this.lblRecentRepositories.Margin = new System.Windows.Forms.Padding(0);
             this.lblRecentRepositories.Name = "lblRecentRepositories";
             this.lblRecentRepositories.Size = new System.Drawing.Size(411, 32);
-            this.lblRecentRepositories.TabIndex = 1;
+            this.lblRecentRepositories.TabIndex = 2;
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.Designer.cs:131]: Please check that the tab order is logical

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs [hunk 7]
```diff
@@ -21,6 +21,7 @@ namespace GitUI.CommandsDialogs.BrowseDialog.DashboardControl
     public partial class UserRepositoriesList : GitExtensionsControl
     {
         private readonly TranslationString _groupRecentRepositories = new("Recent repositories");
+        private readonly TranslationString _repositorySearchPlaceholder = new("Search a repository");
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:24]:
```suggestion
        private readonly TranslationString _repositorySearchPlaceholder = new("Search");
```
or
```suggestion
        private readonly TranslationString _repositorySearchPlaceholder = new("Search repositories");
```

image

I don't think we need a mnemonic though.

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs [hunk 8]
```diff
@@ -96,6 +99,9 @@ public UserRepositoriesList()
 
             DragEnter += OnDragEnter;
             DragDrop += OnDragDrop;
+
+            textBoxSearch.TextChanged += TextBoxSearch_TextChanged;
+            textBoxSearch.KeyDown += TextBoxSearch_KeyDown;
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:104]: Move into the *.designer file

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs [hunk 9]
```diff
@@ -857,7 +891,10 @@ private void OnDragEnter(object sender, DragEventArgs e)
             }
         }
 
-        // returns false only if no repository is selected
+        /// <summary>
+        /// Tries to open the currently selected repository
+        /// </summary>
+        /// <returns>False if no repo is selected, true otherwise</returns>
         private bool TryOpenSelectedRepository()
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:898]: * Modify the signature as follows:
    ```suggestion
            private bool TryOpenSelectedRepository(Repository? repository)
    ```
* Update the existing callsites with `TryOpenSelectedRepository(GetSelectedRepository())`
* Remove `OpenRepository` and use `TryOpenSelectedRepository` instead

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.resx [hunk 10]
```diff
@@ -117,18 +117,18 @@
   <resheader name="writer">
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.resx:117]: Revert changes to this file

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs [hunk 11]
```diff
@@ -633,6 +647,26 @@ private void listView1_MouseClick(object sender, MouseEventArgs e)
             }
         }
 
+        private void TextBoxSearch_TextChanged(object sender, EventArgs e)
+        {
+            ShowRecentRepositories();
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:652]: This is a very handed way of performing the filter. Each time the user will type a character we will be reloading the configuration file, deserializing the list of repositories, recalculating tile sizes and render items.

The filtering must be performed on `recentRepositories` and `favouriteRepositories` collections that are loaded in `ShowRecentRepositories`, adding or removing tiles from the listview as necessary.

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs [hunk 12]
```diff
@@ -8,12 +8,49 @@
...
+    }
+
+    /// <summary>
+    /// Case insensitive text search based on the repository path
+    /// </summary>
+    public class RepositoryTextFilter : IRepositoryFilter
+    {
+        private readonly string _pattern;
+
+        public RepositoryTextFilter(string pattern)
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs:23]: 1. `pattern` is a variable, so it mustn't be set via the .ctor, and pass to `Filter` as an arg instead.
2. In general I'm not convinced this class is needed at all. This functionality can be part of `IUserRepositoriesListController`, e.g. `void IUserRepositoriesListController.FilterRepositories(filter, ListView.ListViewItemCollection repositories)`

**Review by RussKie — CHANGES_REQUESTED** [review]

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs (hunk 11 above)
Comment (davidgiga1993) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:652]: I agree, but one question: The `SplitRcentRepos` may trim the length of the repo list, correct? So would it make sense to keep that list in memory (before the split) and apply the filter on that? This way the search will also find items which are not currently shown in the UI. Or would you expect to only filter the items currently visible?

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.Designer.cs (hunk 6 above)
Comment (davidgiga1993) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.Designer.cs:131]: Yep, the default focus is on the search box, after that comes the recent repo list, then the left menu. The overall behavior stays the same

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs (hunk 11 above)
Comment (RussKie) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:652]: That's a very good Q!
I suppose I could see an expectation to get a result searching for "project" whilst having a repo displayed as `c:\myrepos\so....roject.git` (with the full name `c:\myrepos\somecompany\projects\coolproject.git`). But at the same time I can see why this search result is totally unexpected too.
Personally I'd go for a filter to what's being displayed. That is, if there is not "project" word is visible, then nothing is returned.

@gitextensions/git-extensions-team do you have opnions?

Comment (gerhardol) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:652]: If I understand the question: I would expect to search the full name, not just what happens to be displayed.

Comment (davidgiga1993) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:652]: @RussKie Maybe I maybe formulated the question badly. I meant the number of items shown (defined by `maxRecentRepositories`) might be less than the number of repos known to git extensions.
As I user I would expect the search to search for "any repo known to git extensions matching the patch" rather than "search in the couple of items shown in the UI".
Therefore the search needs to cache the raw list returned by `ThreadHelper.JoinableTaskFactory.Run(() => RepositoryHistoryManager.Locals.LoadRecentHistoryAsync());`.
Do you agree?

Comment (RussKie) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:650]: > RepositoryHistoryManager.Locals.LoadRecentHistoryAsync()

We trim the history size upon save, so the list you get is all we have:
https://github.com/gitextensions/gitextensions/blob/a32fe7a6b79cb3af43fed302a7b548687156d720/GitCommands/UserRepositoryHistory/LocalRepositoryManager.cs#L292-L313

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs [hunk 13]
```diff
@@ -857,28 +889,38 @@ private void OnDragEnter(object sender, DragEventArgs e)
...
 
+            OpenRepository(repository);
+            return true;
+        }
+
+        /// <summary>
+        /// Opens the given repository
+        /// </summary>
+        /// <param name="selected">Repository</param>
+        private void OpenRepository(Repository selected)
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:911]:
```suggestion
        private void OpenRepository(Repository repository)
```

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs (hunk 11 above)
Comment (davidgiga1993) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:650]: Ah I see, thanks for clarifying. So for now I just cache the result from `LoadRecentHistoryAsync()` to avoid loading the data from disk all the time. The search is still applied on the repo list itself.

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs [hunk 14]
```diff
@@ -633,6 +645,26 @@ private void listView1_MouseClick(object sender, MouseEventArgs e)
...
+            if (e.KeyCode == Keys.Enter)
+            {
+                // Open the first repo in the list
+                var items = listView1.Items;
+                if (items.Count == 0)
+                {
+                    return;
+                }
+
+                OpenRepository((Repository)items[0].Tag);
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:664]:
```suggestion
                TryOpenRepository(items[0].Tag as Repository);
```

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs (hunk 13 above)
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:911]: There's no need for the new method.

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs [hunk 15]
```diff
@@ -13,15 +13,21 @@ public interface IUserRepositoriesListController
...
 
     public sealed class UserRepositoriesListController : IUserRepositoriesListController
     {
         private readonly ILocalRepositoryManager _localRepositoryManager;
         private readonly IInvalidRepositoryRemover _invalidRepositoryRemover;
 
+        // Holds the raw, unfiltered list of repositories.
+        // This is done to allow fast filtering of all known repos.
+        private IList<Repository>? _AllRecentRepositories;
+        private IList<Repository>? _AllFavoriteRepositories;
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs:29]:
```suggestion
        private IReadOnlyList<Repository>? _allRecentRepositories;
        private IReadOnlyList<Repository>? _allFavoriteRepositories;
```

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs [hunk 16]
```diff
@@ -340,6 +343,15 @@ void BindRepositories(IReadOnlyList<RecentRepoInfo> repos, bool isFavourite)
             }
         }
 
+        protected override void OnVisibleChanged(EventArgs e)
+        {
+            base.OnVisibleChanged(e);
+
+            // Reset the recent repo cache and the search
+            _controller.ClearCache();
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:351]: Why is this necessary?

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs [hunk 17]
```diff
@@ -85,5 +110,26 @@ public bool IsValidGitWorkingDir(string path)
 
         public bool RemoveInvalidRepository(string path)
            => _invalidRepositoryRemover.ShowDeleteInvalidRepositoryDialog(path);
+
+        private static IList<Repository> Filter(IList<Repository> repositories, string pattern)
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs:114]:
```suggestion
        private static IReadOnlyList<Repository> Filter(IReadOnlyList<Repository> repositories, string pattern)
```

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs [hunk 18]
```diff
@@ -70,11 +85,21 @@ public bool IsValidGitWorkingDir(string path)
                 SortMostRecentRepos = AppSettings.SortMostRecentRepos
             };
 
-            var repositories = ThreadHelper.JoinableTaskFactory.Run(() => RepositoryHistoryManager.Locals.LoadRecentHistoryAsync());
+            if (_AllRecentRepositories == null)
+            {
+                _AllRecentRepositories = ThreadHelper.JoinableTaskFactory.Run(() => RepositoryHistoryManager.Locals.LoadRecentHistoryAsync());
+            }
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs:91]:
```suggestion
            _allRecentRepositories ??= ThreadHelper.JoinableTaskFactory.Run(() => RepositoryHistoryManager.Locals.LoadRecentHistoryAsync());
```

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs (hunk 15 above)
Comment (davidgiga1993) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs:29]: The `LoadFavourite/RecentHistoryAsync` calls return an `IList` not an `IReadOnlyList`

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs (hunk 16 above)
Comment (davidgiga1993) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:351]: The intention was to clear the cache when the view is shown again to update the items shown in case something changed. This solution had a couple of flaws which I just resolved by clearing the cache in the `ShowRecentRepositories` (if required)

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs (hunk 15 above)
Comment (RussKie) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs]: This is a covariant conversion (I think this is the right word), i.e. you can go from `List<T>` to `IReadOnlyList<T>`.

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs [hunk 19]
```diff
@@ -517,7 +533,7 @@ private void UpdateCategoryName(string originalName, string? newName)
         private void contextMenuStrip_Closed(object sender, ToolStripDropDownClosedEventArgs e)
         {
             _rightClickedItem = null;
-            ShowRecentRepositories();
+            ShowRecentRepositories(false);
```
Comment (RussKie) [inline GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesList.cs:536]:
```suggestion
            ShowRecentRepositories(reloadData: false);
```

**Review by RussKie — COMMENTED** [review]
👍 
Few outstanding comments

File: GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs (hunk 15 above)
Comment (davidgiga1993) [reply GitUI/CommandsDialogs/BrowseDialog/DashboardControl/UserRepositoriesListController.cs]: Yes you're right for `List<T>` but the existing methods return an `IList<T>` which does not implement `IReadOnlyList<T>`

**Comment by RussKie** [comment]
Please update English translations and re-submit the pull-request. Refer to https://github.com/gitextensions/gitextensions/wiki/Translations

**Review by RussKie — APPROVED** [review]
👍

**Review by drewnoakes — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you!

**Comment by davidgiga1993** [comment]
Is there already a plan in which version this change will be included?

**Comment by pmiossec** [comment]
There is a milestone to manage the next version:
https://github.com/gitextensions/gitextensions/milestone/60

But I don't think it is up to date.

Because we switched to .net 5 for this version there are things to fix before releasing.
Maybe #9402 (and similar one #9491) should be added.
Fixing the installer need to be done from what I understood.
And some more that are discovered some time to time.

No planned date at the moment.
So if you want to have this feature, you'd better build your own version or use the portable development version built after each PR merged in `master`

**Comment by davidgiga1993** [comment]
Thanks for your replies! Yes I was looking at the milestone and was a bit puzzled that it's missing some stuff, but that explains it 👍
